lib/MusicBrainz/Server/Controller/Doc.pm

```diff
@@ -26,24 +27,31 @@ sub show : Path('')
         return;
     }
 
-    my $bare = $c->req->param('bare') || 0;
-    $c->stash(
-        id => $id,
-        page => $page,
-        google_custom_search => DBDefs->GOOGLE_CUSTOM_SEARCH,
-    );
-
     if ($id =~ /^[^:]+:/i && $id !~ /^Category:/i) {
         $c->response->redirect(sprintf('http://%s/%s', DBDefs->WIKITRANS_SERVER, $id));
         $c->detach;
     }
 
+    my %props = (
+        useGoogleCustomSearch   => boolean_to_json(DBDefs->GOOGLE_CUSTOM_SEARCH),
+        id                      => $id,
+        page                    => $page,
+        wikiServer              => DBDefs->WIKITRANS_SERVER,
+    );
+
     if ($page) {
-        $c->stash->{template} = $bare ? 'doc/bare.tt' : 'doc/page.tt';
-    }
-    else {
+        $c->stash(
+            component_path  => 'doc/DocPage',
+            component_props => \%props,
+            current_view    => 'Node',
+        );
+    }    else {
         $c->response->status(404);
-        $c->stash->{template} = $bare ? 'doc/bare_error.tt' : 'doc/error.tt';
+        $c->stash(
+            component_path  => 'doc/DocError',
+            component_props => \%props,
```

Comment:
"page" and "wikiServer" aren't used here - in fact page will always be undefined. I would like to avoid having two sets of %props though, so hopefully this works fine?